securesystemslib/metadata.py

```diff
@@ -94,3 +89,72 @@ def pae(self) -> bytes:
             len(self.payload),
             self.payload,
         )
+
+    def sign(self, signer: Signer) -> Signature:
+        """Sign the payload and create the signature.
+
+        Arguments:
+            signer: A "Signer" class instance.
+
+        Returns:
+            A "Signature" instance.
+        """
+
+        signature = signer.sign(self.pae())
+        self.signatures.append(signature)
+
+        return signature
+
+    def verify(self, keys: List[Key], threshold: int) -> Set[Key]:
+        """Verify the payload with the provided Keys.
+
+        Arguments:
+            keys: A list of "Key" class instance that is a public key to verify
+                the signatures.
+            threshold: Number of signatures needed to pass the verification.
+
+        Raises:
+            ValueError: If "threshold" is not valid.
+            SignatureVerificationError: If the amount of "recognized_signers" is less
+                than provided threshold.
+
+        Returns:
+            recognized_signers: A set of recognized public keys.
+        """
+
+        recognized_signers = set()
+        pae = self.pae()
+
+        # checks for threshold value.
+        if threshold <= 0:
+            raise ValueError("Threshold must be greater than 0")
+
+        if threshold > len(keys):
+            raise ValueError("Amount of keys must be greater than threshold")
+
+        for signature in self.signatures:
+            for key in keys:
+                # If key and signature include keyIDs but do not match skip.
+                if (
+                    signature.keyid is not None
+                    and key.keyid is not None
+                    and signature.keyid != key.keyid
+                ):
+                    continue
+
+                # If a key verifies the signature, we exit and use the result.
+                if key.verify(signature, pae):
```

Comment:
FYI and because you asked about how to deal with exceptions that might be raised in `Key.verify`:

https://github.com/theupdateframework/python-tuf/pull/1423/commits/414dfc829f34767b80daa15a760a2236334c5a29 discusses returning boolean vs. raising exception in `Key.verify` and opts for the latter.

I suggest we change our `Key` implementations to do the same thing, i.e. return `None` if signature verification passes and raise an exceptions (or rather different exceptions with a common base) in all other cases. Then the caller does not have to case handle True/False/Error but only error type.

I also suggest to do it in a separate PR.